### ISSUE TITLE
Problem in building on CentOS 6.4

### ISSUE THREAD

**Issue by ktk57** [facebook/mysql-5.6#6]
I am trying to build https://github.com/facebook/mysql-5.6 on CentOS 6.4  and the build is failing(I am using devtools-2)

Scanning dependencies of target merge_large_tests-t
[ 82%] Building CXX object unittest/gunit/CMakeFiles/merge_large_tests-t.dir/merge_large_tests.cc.o
In file included from /home/kartik/build_facebook_mysql_5.6/unittest/gunit/merge_large_tests.cc:13:0:
/home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/log_throttle-t.cc: In member function 'virtual void log_throttle_unittest::LogThrottleTest_SlowLogBasic_Test::TestBody()':
/home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/log_throttle-t.cc:84:78: error: invalid conversion from 'bool (_)(THD_, const char_, uint) {aka bool (_)(THD_, const char_, unsigned int)}' to 'bool (_)(THD_, const char_, uint, system_status_var_) {aka bool (_)(THD_, const char_, unsigned int, system_status_var_)}' [-fpermissive]
In file included from /home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/fake_table.h:19:0,
                 from /home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/mock_field_timestamp.h:19,
                 from /home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/copy_info-t.cc:22,
                 from /home/kartik/build_facebook_mysql_5.6/unittest/gunit/merge_large_tests.cc:2:
/home/kartik/mysql-5.6-webscalesql-5.6.16-47/sql/sql_class.h:1469:3: error:   initializing argument 4 of 'Slow_log_throttle::Slow_log_throttle(ulong_, mysql_mutex_t_, ulong, bool (_)(THD_, const char_, uint, system_status_var_), const char_)' [-fpermissive]
In file included from /home/kartik/build_facebook_mysql_5.6/unittest/gunit/merge_large_tests.cc:13:0:
/home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/log_throttle-t.cc: In member function 'virtual void log_throttle_unittest::LogThrottleTest_SlowLogThresholdChange_Test::TestBody()':
/home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/log_throttle-t.cc:122:78: error: invalid conversion from 'bool (_)(THD_, const char_, uint) {aka bool (_)(THD_, const char_, unsigned int)}' to 'bool (_)(THD_, const char_, uint, system_status_var_) {aka bool (_)(THD_, const char_, unsigned int, system_status_var_)}' [-fpermissive]
In file included from /home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/fake_table.h:19:0,
                 from /home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/mock_field_timestamp.h:19,
                 from /home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/copy_info-t.cc:22,
                 from /home/kartik/build_facebook_mysql_5.6/unittest/gunit/merge_large_tests.cc:2:
/home/kartik/mysql-5.6-webscalesql-5.6.16-47/sql/sql_class.h:1469:3: error:   initializing argument 4 of 'Slow_log_throttle::Slow_log_throttle(ulong_, mysql_mutex_t_, ulong, bool (_)(THD_, const char_, uint, system_status_var_), const char_)' [-fpermissive]
In file included from /home/kartik/build_facebook_mysql_5.6/unittest/gunit/merge_large_tests.cc:13:0:
/home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/log_throttle-t.cc: In member function 'virtual void log_throttle_unittest::LogThrottleTest_SlowLogSuppressCount_Test::TestBody()':
/home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/log_throttle-t.cc:154:78: error: invalid conversion from 'bool (_)(THD_, const char_, uint) {aka bool (_)(THD_, const char_, unsigned int)}' to 'bool (_)(THD_, const char_, uint, system_status_var_) {aka bool (_)(THD_, const char_, unsigned int, system_status_var_)}' [-fpermissive]
In file included from /home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/fake_table.h:19:0,
                 from /home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/mock_field_timestamp.h:19,
                 from /home/kartik/mysql-5.6-webscalesql-5.6.16-47/unittest/gunit/copy_info-t.cc:22,
                 from /home/kartik/build_facebook_mysql_5.6/unittest/gunit/merge_large_tests.cc:2:
/home/kartik/mysql-5.6-webscalesql-5.6.16-47/sql/sql_class.h:1469:3: error:   initializing argument 4 of 'Slow_log_throttle::Slow_log_throttle(ulong_, mysql_mutex_t_, ulong, bool (_)(THD_, const char_, uint, system_status_var_), const char_)' [-fpermissive]
At global scope:
cc1plus: warning: unrecognized command line option "-Wno-null-dereference" [enabled by default]
make[2]: *_\* [unittest/gunit/CMakeFiles/merge_large_tests-t.dir/merge_large_tests.cc.o] Error 1
make[1]: **\* [unittest/gunit/CMakeFiles/merge_large_tests-t.dir/all] Error 2
make: **\* [all] Error 2

Am I missing something? Please let me know if there is a mailing list.
I basically need a non-blocking mysqlclient(C bindings).

Thanks for your patience !

**Comment by steaphan-fb-com** [comment]
I hope you've managed to resolve this in the meantime.

If not, I suggest you try to use the most recent branch, as the one used here is actually pretty old at this point.  If that doesn't work, please open another issue with the details, and I will take a closer look.

Sorry for my delayed response on this one.

**Comment by AllenChien** [comment]
I encounter below compile errors, code base is cloned from latest github.
1. CentOS release 6.4 (Final)
2. gcc version 4.8.2 (GCC)

[ 63%] Building C object vio/CMakeFiles/vio.dir/viosslfactories.c.o
/data1/mysql/mysql-5.6-webscalesql-5.6.23.71/vio/viosslfactories.c: In function ‘new_VioSSLFd’:
/data1/mysql/mysql-5.6-webscalesql-5.6.23.71/vio/viosslfactories.c:265:3: warning: implicit declaration of function ‘ERR_clear_error’ [-Wimplicit-function-declaration]
   ERR_clear_error();
   ^
/data1/mysql/mysql-5.6-webscalesql-5.6.23.71/vio/viosslfactories.c:306:5: error: unknown type name ‘EC_KEY’
     EC_KEY _ecdh = EC_KEY_new_by_curve_name(NID_X9_62_prime256v1);
     ^
/data1/mysql/mysql-5.6-webscalesql-5.6.23.71/vio/viosslfactories.c:306:5: warning: implicit declaration of function ‘EC_KEY_new_by_curve_name’ [-Wimplicit-function-declaration]
/data1/mysql/mysql-5.6-webscalesql-5.6.23.71/vio/viosslfactories.c:306:45: error: ‘NID_X9_62_prime256v1’ undeclared (first use in this function)
     EC_KEY *ecdh = EC_KEY_new_by_curve_name(NID_X9_62_prime256v1);
                                             ^
/data1/mysql/mysql-5.6-webscalesql-5.6.23.71/vio/viosslfactories.c:306:45: note: each undeclared identifier is reported only once for each function it appears in
/data1/mysql/mysql-5.6-webscalesql-5.6.23.71/vio/viosslfactories.c:308:7: warning: implicit declaration of function ‘SSL_CTX_set_tmp_ecdh’ [-Wimplicit-function-declaration]
       SSL_CTX_set_tmp_ecdh(ssl_fd->ssl_context, ecdh);
       ^
/data1/mysql/mysql-5.6-webscalesql-5.6.23.71/vio/viosslfactories.c:309:7: warning: implicit declaration of function ‘EC_KEY_free’ [-Wimplicit-function-declaration]
       EC_KEY_free(ecdh);
       ^
make[2]: *_\* [vio/CMakeFiles/vio.dir/viosslfactories.c.o] Error 1
make[1]: **\* [vio/CMakeFiles/vio.dir/all] Error 2
make: **\* [all] Error 2

**Comment by AllenChien** [comment]
cmake with -DWITH_SSL=system, everything goes fine!
